File: app/src/main/java/org/michaelbel/moviemade/ui/base/BaseMvp.java
package org.michaelbel.moviemade.ui.base;

import androidx.fragment.app.Fragment;
import android.view.View;

public interface BaseMvp {

    void startFragment(Fragment fragment, View container);

    void startFragment(Fragment fragment, int containerId);

    void startFragment(Fragment fragment, View container, String tag);

    void startFragment(Fragment fragment, int containerId, String tag);

    void finishFragment();
}

File: app/src/main/java/org/michaelbel/moviemade/ui/base/BaseMvp.kt
package org.michaelbel.moviemade.ui.base

import android.view.View
import androidx.fragment.app.Fragment

interface BaseMvp {

    fun startFragment(fragment: Fragment, container: View)

    fun startFragment(fragment: Fragment, containerId: Int)

    fun startFragment(fragment: Fragment, container: View, tag: String)

    fun startFragment(fragment: Fragment, containerId: Int, tag: String)

    fun finishFragment()
}